src/listener.coffee

```diff
@@ -39,7 +39,7 @@ class TextListener extends Listener
   # callback - A Function that is triggered if the incoming message matches.
   constructor: (@robot, @regex, @callback) ->
     @matcher = (message) =>
-      if message instanceof TextMessage
+      if message.match
```

Comment:
It's just a duck typing, at the end is the same. I can change it if you are concerned about non text messages with `match` properties.